src/langchain_google_bigtable/document_loader.py

```diff
@@ -0,0 +1,273 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import struct
+import uuid
+from enum import Enum
+from typing import Any, Callable, Iterator, List, Optional
+
+from google.cloud import bigtable
+from langchain_community.document_loaders.base import BaseLoader
+from langchain_core.documents import Document
+
+COLUMN_FAMILY = "langchain"
+CONTENT_COLUMN_NAME = "content"
+ID_METADATA_KEY = "rowkey"
+
+
+class Encoding(Enum):
+    UTF8 = "utf-8"
+    UTF16 = "utf-16"
+    ASCII = "ascii"
+    INT_LITTLE_ENDIAN = "int_little_endian"
+    INT_BIG_ENDIAN = "int_big_endian"
+    DOUBLE = "double"
+    FLOAT = "float"
+    BOOL = "bool"
+    CUSTOM = "custom"
+
+
+def _not_implemented(_: Any) -> Any:
+    raise NotImplementedError(
+        "decoding/encoding function not set for custom encoded metadata key"
+    )
+
+
+class MetadataMapping:
+    def __init__(
+        self,
+        column_family: str,
+        column_name: str,
+        metadata_key: str,
+        encoding: Encoding,
+        custom_encoding_func: Callable[[Any], bytes] = _not_implemented,
+        custom_decoding_func: Callable[[bytes], Any] = _not_implemented,
+    ):
+        self.column_family = column_family
+        self.column_name = column_name
+        self.metadata_key = metadata_key
+        self.encoding = encoding
+        self.custom_encoding_func = custom_encoding_func
+        self.custom_decoding_func = custom_decoding_func
```

Comment:
nit: could we just use a dataclass here? https://docs.python.org/3/library/dataclasses.html

Reply:
Done